src/dataExplorer/components/FluxQueryBuilder.tsx

```diff
@@ -1,38 +1,105 @@
 import React, {FC, useState} from 'react'
 
 // Components
-import {DraggableResizer, Orientation} from '@influxdata/clockface'
+import {
+  AlignItems,
+  DraggableResizer,
+  FlexBox,
+  FlexDirection,
+  Orientation,
+} from '@influxdata/clockface'
 import {QueryProvider} from 'src/shared/contexts/query'
+import {EditorProvider} from 'src/shared/contexts/editor'
 import {ResultsProvider} from 'src/dataExplorer/components/ResultsContext'
 import ResultsPane from 'src/dataExplorer/components/ResultsPane'
+import SidePane from 'src/dataExplorer/components/SidePane'
 import Schema from 'src/dataExplorer/components/Schema'
 
 // Styles
 import './FluxQueryBuilder.scss'
+import LeftArrow from 'assets/images/collapse-left.svg'
+import RightArrow from 'assets/images/collapse-right.svg'
 
-const INITIAL_VERT_RESIZER_HANDLE = 0.2
+const INITIAL_LEFT_VERT_RESIZER_HANDLE = 0.2
+const INITIAL_RIGHT_VERT_RESIZER_HANDLE = 0.8
 
 const FluxQueryBuilder: FC = () => {
   const [vertDragPosition, setVertDragPosition] = useState([
-    INITIAL_VERT_RESIZER_HANDLE,
+    INITIAL_LEFT_VERT_RESIZER_HANDLE,
+    INITIAL_RIGHT_VERT_RESIZER_HANDLE,
   ])
+  const [showLeft, toggleLeftPanel] = useState(true)
+  const [showRight, toggleRightPanel] = useState(true)
+
+  const adjustMiddlePanel = grow => {
+    if (!grow) {
+      return
+    }
+    const el = document.querySelector(
+      '[data-testid="flux-query-builder-middle-panel"]'
+    )
+    if (el) {
+      (el as any).style.flexGrow = 0.8
+    }
```

Comment:
I am not a huge fan of this approach. But the problem is mixing the DraggableResizer (which sets the flex-basis on all the panels), with a collapsing button for the leftPanel and rightPanel.

Without this manually triggering of the central panel to flexGrow, it was not expanding to fill the space:

https://user-images.githubusercontent.com/10232835/172515928-446a6ed2-e4f4-4c85-834a-500033c70f85.mov

Reply:
Surely there's a better way